middleware/authorization.js

```diff
@@ -0,0 +1,62 @@
+const pool = require("../db");
+const utils = require('../utils');
+
+function verifyRole(role_name) {
+    return async (req, res, next) => {
+        try {
+            // todo: get user id from request (token?)
+            const user_id = 1; // temp
```

Comment:
you're correct :) this can be used to check any role, for now we verify mentor role to get all applications, but later we will 100% have endpoints that will need member or other role verification